Source/Core/Common/GL/GLUtil.cpp

```diff
@@ -35,22 +36,23 @@ GLuint OpenGL_CompileProgram(const char* vertexShader, const char* fragmentShade
 
 	if (Result && stringBufferUsage)
 	{
-		ERROR_LOG(VIDEO, "GLSL vertex shader warnings:\n%s%s", stringBuffer, vertexShader);
+		ERROR_LOG(VIDEO, "GLSL vertex shader warnings:\n%s%s", stringBuffer, vertexShader.c_str());
 	}
 	else if (!Result)
 	{
-		ERROR_LOG(VIDEO, "GLSL vertex shader error:\n%s%s", stringBuffer, vertexShader);
+		ERROR_LOG(VIDEO, "GLSL vertex shader error:\n%s%s", stringBuffer, vertexShader.c_str());
 	}
 	else
 	{
-		DEBUG_LOG(VIDEO, "GLSL vertex shader compiled:\n%s", vertexShader);
+		DEBUG_LOG(VIDEO, "GLSL vertex shader compiled:\n%s", vertexShader.c_str());
 	}
 
 	bool shader_errors = !Result;
 #endif
 
 	// compile fragment shader
-	glShaderSource(fragmentShaderID, 1, &fragmentShader, nullptr);
+	shader = fragmentShader.c_str();
+	glShaderSource(fragmentShaderID, 1, &shader, nullptr);
```

Comment:
This can just be `glShaderSource(fragmentShaderID, 1, &shader[0], nullptr);` or `glShaderSource(fragmentShaderID, 1, shader.data(), nullptr);`, no? Same with the other similar usage above.